azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1761,6 +1761,20 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
       throws ServletException, IOException {
     final User user = session.getUser();
     final String projectName = (String) multipart.get("project");
+
+    // Fetch the uploader's IP
+    String uploaderIPAddr = "";
+    if (req != null) {
+      uploaderIPAddr = req.getHeader("X-FORWARDED-FOR");
+      if (uploaderIPAddr == null || uploaderIPAddr.isEmpty()) {
+        logger.debug("Failed to fetch remote Address using \"X-FORWARDED-FOR\"");
+        uploaderIPAddr = req.getRemoteAddr();
+      }
+      logger.info("uploaderIPAddr = " + uploaderIPAddr);
+    } else {
+      logger.info("HttpServletRequest is NULL");
```

Comment:
These log messages are not very useful as they don't include information about the specific request/project/user

Reply:
Good point. I added them for dev purpose and never improved on them.